From ac574287e1ca12428c0e8d838c84c01064811fd5 Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Thu, 2 Feb 2023 15:03:50 +0100
Subject: [PATCH] Adjust IDL terms in Web Animations

Some terms are partially re-defined in Level 2. Parts of the patch that update
these definitions are only needed because Level 2 is a delta spec. They will
likely need to be kept around for as long as that remains the case.
---
 ed/idl/web-animations.idl | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/ed/idl/web-animations.idl b/ed/idl/web-animations.idl
index 1b078e6ea..956d700f4 100644
--- a/ed/idl/web-animations.idl
+++ b/ed/idl/web-animations.idl
@@ -5,7 +5,6 @@
 
 [Exposed=Window]
 interface AnimationTimeline {
-    readonly attribute double? currentTime;
 };
 
 dictionary DocumentTimelineOptions {
@@ -24,8 +23,6 @@ interface Animation : EventTarget {
              attribute DOMString                id;
              attribute AnimationEffect?         effect;
              attribute AnimationTimeline?       timeline;
-             attribute double?                  startTime;
-             attribute double?                  currentTime;
              attribute double                   playbackRate;
     readonly attribute AnimationPlayState       playState;
     readonly attribute AnimationReplaceState    replaceState;
@@ -58,12 +55,9 @@ interface AnimationEffect {
 };
 
 dictionary EffectTiming {
-    double                             delay = 0;
-    double                             endDelay = 0;
     FillMode                           fill = "auto";
     double                             iterationStart = 0.0;
     unrestricted double                iterations = 1.0;
-    (unrestricted double or DOMString) duration = "auto";
     PlaybackDirection                  direction = "normal";
     DOMString                          easing = "linear";
 };
@@ -84,9 +78,6 @@ enum FillMode { "none", "forwards", "backwards", "both", "auto" };
 enum PlaybackDirection { "normal", "reverse", "alternate", "alternate-reverse" };
 
 dictionary ComputedEffectTiming : EffectTiming {
-    unrestricted double  endTime;
-    unrestricted double  activeDuration;
-    double?              localTime;
     double?              progress;
     unrestricted double? currentIteration;
 };
@@ -156,14 +147,3 @@ partial interface mixin DocumentOrShadowRoot {
 };
 
 Element includes Animatable;
-
-[Exposed=Window]
-interface AnimationPlaybackEvent : Event {
-    constructor(DOMString type, optional AnimationPlaybackEventInit eventInitDict = {});
-    readonly attribute double? currentTime;
-    readonly attribute double? timelineTime;
-};
-dictionary AnimationPlaybackEventInit : EventInit {
-    double? currentTime = null;
-    double? timelineTime = null;
-};
-- 
2.37.1.windows.1

